roles/network_plugin/cilium/tasks/install.yml

```diff
@@ -33,13 +33,14 @@
     - {name: cilium, file: cilium-config.yml, type: cm}
     - {name: cilium, file: cilium-crb.yml, type: clusterrolebinding}
     - {name: cilium, file: cilium-cr.yml, type: clusterrole}
-    - {name: cilium, file: cilium-secret.yml, type: secret, when: cilium_ipsec_enabled}
+    - {name: cilium, file: cilium-secret.yml, type: secret}
     - {name: cilium, file: cilium-ds.yml, type: ds}
     - {name: cilium, file: cilium-deploy.yml, type: deploy}
     - {name: cilium, file: cilium-sa.yml, type: sa}
   register: cilium_node_manifests
   when:
     - inventory_hostname in groups['kube_control_plane']
+    - item.file != "cilium-secret.yml" or (item.file == "cilium-secret.yml" and cilium_ipsec_enabled)
```

Comment:
I think you can write it as:
```
item.when is defined and item.when
```

Reply:
thx, this didn't seem to work for me

Reply:
You've put back the `when:` that you're removing in this commit ?

Reply:
yes, I set the item back to:

```
- {name: cilium, file: cilium-secret.yml, type: secret, when: cilium_ipsec_enabled}
```

when I tested

Reply:
Sorry my logic is wrong
`not item.when is defined or item.when` ?

Reply:
thx, that produces the error:

failed: [k8m01] (item={'name': 'cilium', 'file': 'cilium-secret.yml', 'type': 'secret', 'when': 'cilium_ipsec_enabled'}) => {"ansible_loop_var": "item", "changed": false, "item": {"file": "cilium-secret.yml", "name": "cilium", "type": "secret", "when": "cilium_ipsec_enabled"}, "msg": "AnsibleUndefinedVariable: 'cilium_ipsec_key' is undefined"}

Reply:
Simpler and should be working this time:
`item.when | default(true)`

Reply:
unfortunately this still produces the error:

```

Reply:
Make sure `cilium_ipsec_key` is defined in the role defaults ?

Reply:
That would work but I didn't make the original IPsec PR. I assume the person who did has a reason for not defining a default for cilium_ipsec_key.

Reply:
Sorry I was quickly answering from my phone, would `{name: cilium, file: cilium-secret.yml, type: secret, when: cilium_ipsec_enabled is defined }` work then ?

Reply:
Sorry for lagging behind but I was thinking of fixing it the same way @champtar wrote.
Weird that it's not working 🤔